### PR TITLE
Set CMAKE_MODULE_PATH appropriately to always use preferred linalg-modules

### PR DESCRIPTION
**Pull request by ajaypanyala**

### CODE REVIEW

File: CMakeLists.txt
Comment (wavefunction91) [inline CMakeLists.txt:9]: Isn't this equivalent to `list( PREPEND`?

File: src/CMakeLists.txt
Comment (wavefunction91) [inline src/CMakeLists.txt:59]: I would change this to
```
if( GAUXC_ENABLE_OPENMP )
  target_link_libraries( gauxc PUBLIC OpenMP::OpenMP_CXX )
endif()
```

**Review by wavefunction91 — CHANGES_REQUESTED** [review]

**Comment by wavefunction91** [comment]
@ajaypanyala Is there anything else you want to add to this? I'm going to hold off on merging until https://github.com/wavefunction91/linalg-cmake-modules/pull/1 is merged

**Comment by ajaypanyala** [comment]
I am done with my edits. Thanks!

File: cmake/gauxc-dep-versions.cmake
Comment (ajaypanyala) [inline cmake/gauxc-dep-versions.cmake:1]: Note before merging: This needs to change once [wavefunction91/linalg-cmake-modules#1](https://github.com/wavefunction91/linalg-cmake-modules/pull/1) is merged

File: cmake/gauxc-dep-versions.cmake
Comment (ajaypanyala) [inline cmake/gauxc-dep-versions.cmake:2]: TODO: Pin commit before merging.

**Pull request closed without merging**
